### PR TITLE
feat: Add singularity module

### PR DESCRIPTION
**Pull request by ntadej**
#### Description

Singularity is one of the tools to run containers. A popular use case is to run it as a development environment, especially for older platforms.

#### Motivation and Context

As one might use several different images it is useful to display the image we are currently in.

Currently the default display is after hostname wrapped in `[` and `]` as it is the shortest. I'm open to suggestions if another defaults are more reasonable.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Screenshots (if appropriate):
Screenshot 2019-10-28 at 23 26 39

#### How Has This Been Tested?

- [ ] I have tested using **MacOS**
- [x] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

### CODE REVIEW

File: src/configs/singularity.rs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+use crate::config::{ModuleConfig, RootModuleConfig, SegmentConfig};
+
+use ansi_term::{Color, Style};
+use starship_module_config_derive::ModuleConfig;
+
+#[derive(Clone, ModuleConfig)]
+pub struct SingularityConfig<'a> {
+    pub symbol: Option<SegmentConfig<'a>>,
```
Comment (heyrict) [inline src/configs/singularity.rs]: I think using `symbol: SegmentConfig` with `""` as default value should be good enough, as long as you don't render the prompt conditionally.

File: src/configs/singularity.rs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+impl<'a> RootModuleConfig<'a> for SingularityConfig<'a> {
+    fn new() -> Self {
+        SingularityConfig {
+            symbol: None,
+            label: "",
+            prefix: "[",
+            suffix: "]",
+            style: Color::Blue.bold().dimmed(),
+            disabled: true,
```
Comment (heyrict) [inline src/configs/singularity.rs]: You need to explicitly add code to `modules/singularity.rs` to disable the module. Currently there is no high order approach for starship to judge the disabled option when the module is disabled as a default.

**Comment by matchai** [comment]
Closing this PR as stale.
Feel free to reopen it when you get around to addressing the review comments.

**Comment by ntadej** [comment]
Sorry for delaying this I rebased and fixed the comments. Let's see if I can figure out how to open this PR.

File: src/configs/singularity.rs (hunk 2 above)
Comment (ntadej) [reply src/configs/singularity.rs]: Done.

File: src/configs/singularity.rs (hunk 1 above)
Comment (ntadej) [reply src/configs/singularity.rs]: Done.

**Comment by matchai** [comment]
Unfortunately, I no longer can since the branch was force-pushed or recreated. 😅 
Feel free to open a new PR.

**Pull request closed without merging**